### PR TITLE
feat: Gas estimate UI + Loading UI updates: Version A (WIP)

### PR DESCRIPTION
**Pull request by ianlapham**
- Version with router details in hover
- implements #2814

### CODE REVIEW

File: src/hooks/useUSDCPrice.ts [hunk 1]
```diff
@@ -65,3 +70,54 @@ export function useUSDCValue(currencyAmount: CurrencyAmount<Currency> | undefine
...
+  } catch (error) {
+    return undefined
+  }
+}
+
+/**
+ * Get gas cost estimate for a dummy ETH / Stable v3 trade
+ * @returns Estimate in form of stablecoin amount or null
+ */
+export function useDefaultGasCostEstimate(): {
```
Comment (justindomingue) [inline src/hooks/useUSDCPrice.ts:102]: feels like this function should be in a separate file `src/hooks/gas.ts` or something

Comment (ianlapham) [reply src/hooks/useUSDCPrice.ts:102]: Yeah thought of this too - might move

File: src/state/routing/useRoutingAPITrade.ts [hunk 2]
```diff
@@ -59,7 +60,11 @@ export function useRoutingAPITrade<TTradeType extends TradeType>(
   tradeType: TTradeType,
   amountSpecified?: CurrencyAmount<Currency>,
   otherCurrency?: Currency
-): { state: V3TradeState; trade: Trade<Currency, Currency, TTradeType> | null } {
+): {
+  state: V3TradeState
+  trade: Trade<Currency, Currency, TTradeType> | null
+  gasUseEstimateUSD: CurrencyAmount<Token> | null // dollar amount in active chains stabelcoin
```
Comment (justindomingue) [inline src/state/routing/useRoutingAPITrade.ts:66]: have you considered making that a part of `Trade`? like create a new `Trade` object in the interface that support gas estimates so it's more easily available throughout the app

`class InterfaceTrade extends Trade {
    private gasEstimate: CurrencyAmount<> | null
}
`

Comment (justindomingue) [reply src/state/routing/useRoutingAPITrade.ts:73]: don't

**Pull request closed without merging**